src/env/tock/clock.rs

```diff
@@ -38,12 +39,32 @@ fn wrapping_sub_u24(lhs: usize, rhs: usize) -> usize {
     lhs.wrapping_sub(rhs) & 0xffffff
 }
 
+/// Clock that produces timers through Tock syscalls.
+///
+/// To guarantee correctness, you have to call any of its functions at least once per full tick
+/// counter wrap. In our case, 24 bit ticks with a 32 kHz frequency wrap after 512 seconds. If you
+/// can't guarantee to regularly create or check timers, call tickle at least every 8 minutes.
 #[derive(Default)]
 pub struct TockClock {
     epoch: usize,
     tick: usize,
 }
 
+impl TockClock {
+    /// Elapses timers before the clock wraps.
+    ///
+    /// Call this regularly to timeout reliably despite wrapping clock ticks.
+    pub fn tickle(&mut self) {
+        let cur_tick = syscalls::command(DRIVER_NUMBER, command_nr::GET_CLOCK_VALUE, 0, 0)
+            .ok()
+            .unwrap();
+        if wrapping_sub_u24(self.tick, cur_tick) < 0x80_0000 {
```

Comment:
If the documentation about 512 seconds is true, then you can just do `cur_tick < self.tick` here.

Reply:
Indeed!